akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -90,14 +94,22 @@ private[cluster] class ClusterRemoteWatcher(
   def memberRemoved(m: Member, previousStatus: MemberStatus): Unit =
     if (m.address != selfAddress) {
       clusterNodes -= m.address
-      // The reason we don't quarantine gracefully removed members (leaving) is that
-      // Cluster Singleton need to exchange TakeOver/HandOver messages.
+
       if (previousStatus == MemberStatus.Down) {
         quarantine(m.address, Some(m.uniqueAddress.uid), s"Cluster member removed, previous status [$previousStatus]")
+      } else if (arteryEnabled) {
+        // don't quarantine gracefully removed members (leaving) directly,
+        // give Cluster Singleton some time to exchange TakeOver/HandOver messages.
+        import context.dispatcher
+        context.system.scheduler.scheduleOnce(cluster.downingProvider.downRemovalMargin, self, DelayedQuarantine(m, previousStatus))
```

Comment:
I have created a new issue for redesigning the cluster leaving process, but that's a rather big thing. Until then we need something simple that can work, and I think a simple delay like this should be enough for now.

We already have enough test coverage, singleton tests will fail.

We can't use `downRemovalMargin`. It's by default zero. We should use something like 20 - 30 seconds. It doesn't feel great to introduce a new config property for this, but hardcoded value is not so nice either.